projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules.php

```diff
@@ -102,6 +103,40 @@ public function register_routes() {
 			)
 		);
 
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<schedule_id>[\w]+)/logs',
+			array(
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'add_log' ),
+					'permission_callback' => array( $this, 'add_log_permissions_check' ),
+					'args'                => array(
+						'action'  => array(
+							'description' => 'The action to be logged',
+							'type'        => 'string',
+							'required'    => true,
+						),
+						'message' => array(
+							'description' => 'The message to be logged',
+							'type'        => 'string',
+							'required'    => false,
+						),
+						'context' => array(
+							'description' => 'The context to be logged',
+							'type'        => 'object',
+							'required'    => false,
```

Comment:
Is there an opportunity for us to define out more of the object shape? Or would that make it harder to work with the endpoint?

Reply:
I wanted this to be as _anything_ as possible, because context is just metadata and can really take any shape. We could define the possible shape(s) but I think this will make future additions harder.

Reply:
That makes sense, thanks!